cfdm/test/test_Data.py

```diff
@@ -68,6 +68,12 @@ def test_Data__repr__str(self):
             _ = repr(d)
             _ = str(d)
 
+        # Test when the data contains date-times with the first
+        # element masked
+        dt = numpy.ma.array([10, 20], mask=[True, False])
+        d = cfdm.Data(dt, units="days since 2000-01-01")
+        self.assertTrue(str(d) == "[--, 2000-01-21 00:00:00]")
```

Comment:
Since the `repr` was also failing, might be good to also check that for robustness (I realise it probably calls `str()` in some way under-the-hood for the final component of the output, hence the corresponding failure, but for a one-liner perhaps why not)?

```suggestion
        self.assertTrue(str(d) == "[--, 2000-01-21 00:00:00]")
        self.assertTrue(repr(d) == "<Data(2): [--, 2000-01-21 00:00:00]>")
```